src/pool/srv_pool.c

```diff
@@ -5987,12 +5987,25 @@ __ds_pool_mark_upgrade_completed(uuid_t pool_uuid, struct pool_svc *svc, int rc)
 	if (rc1)
 		D_GOTO(out_tx, rc1);
 
+	if (rc != 0) {
+		/*
+		 * Currently, the upgrade global version may have not been updated yet, if
+		 * pool_upgrade_props has encountered an error.
+		 */
+		d_iov_set(&value, &global_version, sizeof(global_version));
```

Comment:
No change requested. Question (I must have forgotten how interop/upgrade is supposed to work) - is there logic anywhere that enforces an upgrade of one layout version at a time? For example, before an upgrade attempt, if we have a pool with global_version=2, upgrade_version=2, and the software installation has DAOS_POOL_VERSION=4 - does the upgrade code just try to upgrade the pool to the latest DAOS_POOL_VERSION(4), or is some logic in place to force an upgrade to global version 2  first, then another upgrade to global version 3?

Reply:
I can't find any code that forces the example pool to be upgraded first to 3 then to 4. @wangshilong, do you know the answer?

Reply:
We don't support jump upgrade,  that means admin supposed to upgrade to 3 firstly and then upgrade to 4. Currently codes might just work as it is, eg if there are missing properties, just create new one, but we I think interop tests never test that too. Maybe we'd better add a check in the upgrade codes to reject that kind of upgrade.

Reply:
Create a PR to following this: https://github.com/daos-stack/daos/pull/15919